storage-access.bs

```diff
@@ -65,53 +69,77 @@ This specification depends on the Infra standard. [[!INFRA]]
 
-Later on, Alex visits `https://video.example/`, which has an <{iframe}> on it which loads `https://social.example/heart-button-widget`. In this case, the `social.example` {{Document}} |doc| is [=third party=], and the cookie set previously might or might not be visible from |doc|`.`{{Document/cookie}}, depending on User Agent storage access policies.
+Later on, Alex visits `https://video.example/`, which has an <{iframe}> on it which loads `https://social.example/heart-button`. In this case, the `social.example` {{Document}} |doc| is in a [=third party context=], and the cookie set previously might or might not be visible from |doc|`.`{{Document/cookie}}, depending on User Agent storage access policies.
 
 Script in the <{iframe}> can call |doc|`.`{{Document/hasStorageAccess()}} to determine if it has access to the cookie. If it does not have access, it can request access by calling |doc|`.`{{Document/requestStorageAccess()}}.
 
 </div>
 
+<dfn>First party data</dfn> is client-side storage that would be available to a [=site=] were it loaded in a [=first-party-site context=].
+
+A {{Document}} is in a <dfn>first-party-site context</dfn> if it is the [=active document=] of a [=top-level browsing context=]. Otherwise, it is in a [=first-party-site context=] if it is an [=active document=] and the [=environment settings object/origin=] and [=top-level origin=] of its [=relevant settings object=] are [=same site=] with one another.
+
+A {{Document}} is in a <dfn>third party context</dfn> if it is not in a [=first-party-site context=].
+
+ISSUE(10): If we let nested <{iframe}>s use this API, we may have to revisit these definitions.
+
 <h3 id="ua-state">User Agent state related to storage access</h3>
 
-Each User Agent maintains a <dfn>storage access map</dfn>, an [=ordered map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
+A <dfn>storage access map</dfn> is a [=map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
 
-ISSUE: What's the lifetime of the [=storage access map=] and of its entries?
+User Agents maintain a single <dfn>global storage access map</dfn>.
 
-ISSUE(28): Tie the [=storage access map=] to agent clusters.
+ISSUE(2): What is the lifecycle of the [=global storage access map=]? How long do we remember its contents? Firefox and Safari differ here.
 
-A <dfn>partitioned storage key</dfn> is a [=pair=] consisting of a top-level site and a site, both [=sites=].
+ISSUE(5): When do we age out entries in the [=global storage access map=]? See also [Scope of Storage Access](https://github.com/privacycg/storage-access#scope-of-storage-access).
 
-ISSUE: Replace this [=partitioned storage key=] concept with whatever Anne comes up with for [[!STORAGE]].
+Each [=agent cluster=] has a <dfn for="agent cluster">storage access map</dfn>.
```

Comment:
The one weird thing here is that the map goes from site to flags, but an agent cluster is already keyed on a site so it would really only need the flags.

Another alternative would be to put the cache on the browsing context group or some such. I probably need a more detailed discussion with @bakulf to see what would be good here.

Reply:
@annevk: The global storage access map uses (site, site) keys, and it seemed easiest for the agent cluster ones to use keys of the same type. I'd be okay with changing that, though, maybe in a followup?

Reply:
Yeah, I think this needs some further consideration. Another oddity is that agent clusters are subject to GC so if a browsing context group could not contain an agent cluster for a while (site was navigated away from, frame got removed) and then it's needed again it would depend on GC whether that agent cluster has the cached state.